src/audio.rs

```diff
@@ -14,18 +15,75 @@ pub enum AudioCommand {
     Resume,
 }
 
+pub(crate) struct PlayAudioCommandArgs {
+    /// The settings for this Play command.
+    pub(crate) settings: PlayAudioSettings,
+
+    /// An instance handle to communicate with the consumer.
+    pub(crate) instance_handle: InstanceHandle,
+}
+
 pub enum AudioCommandResult {
     Ok,
     Retry,
 }
 
 #[derive(PartialEq, Eq, Hash, Clone)]
-pub struct PlayAudioSettings {
+pub(crate) struct PlayAudioSettings {
     pub source: Handle<AudioSource>,
     pub intro_source: Option<Handle<AudioSource>>,
     pub looped: bool,
 }
 
+/// Allows you to interact with a playing sound.
+#[derive(Clone, Debug, Eq, Hash, PartialEq)]
+pub struct InstanceHandle {
+    id: u64,
+}
+
+static NEXT_INSTANCE_HANDLE_ID: AtomicU64 = AtomicU64::new(0);
+
+impl InstanceHandle {
+    fn new() -> InstanceHandle {
+        let id = NEXT_INSTANCE_HANDLE_ID.fetch_add(1, Ordering::SeqCst);
+        InstanceHandle { id }
+    }
+}
+
+/// Information about a currently playing sound.
+#[derive(Clone, Debug)]
+pub struct PlaybackState {
+    pub(crate) status: PlaybackStatus,
+    pub(crate) position: Option<f64>,
+}
+
+impl PlaybackState {
+    /// Return the PlaybackStatus of this instance
+    pub fn status(&self) -> PlaybackStatus {
+        self.status.clone()
+    }
+
+    /// Return the playback position of this instance
+    pub fn position(&self) -> Option<f64> {
+        self.position.clone()
+    }
+}
+
+/// Playback status of a currently playing sound.
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub enum PlaybackStatus {
```

Comment:
I would probably try and refactor this a bit (but that can happen after the merge!): I think we are missing a distinction between "Sound will start playing soon" and "instance was removed"/"is stopped". It might also be nicer to have the position in the enums. That should allow us to get rid of the Option. "Will play soon" and "Stopped" probably have no position, but all other states have one.

But I am fine with merging the current state. It might be helpful to play around with this a bit and then try some refactoring.

Reply:
I'll resolve this. We can iterate on the exact states.

Reply:
> But I am fine with merging the current state. It might be helpful to play around with this a bit and then try some refactoring.

Ok, sure. 

Would we set the status to "Will play soon" in Audio::play() and similar? How about Initializing for the enum state?

Reply:
Initializing would work too. Not sure about the best naming there.